swarm/api/routes/evolution.py

```
@@ -223,8 +225,14 @@ async def get_run_evolution_patches(run_id: str):
         PendingPatchesResponse with patches for this run.
 
     Raises:
+        400: Path traversal detected.
         404: Run not found or no wisdom outputs.
     """
+    try:
+        validate_path_component(run_id, "run_id")
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)